walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,179 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.Comparator;
+
 
 public final class FindMeetingQuery {
+
+
+
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+        
+        ArrayList<TimeRange> availableTimes = new ArrayList();
+        
+        ArrayList<Event> eventsCopy = new ArrayList<Event>();
+
+        eventsCopy = loadEvents(events);
+        
+        eventsCopy.sort(Comparator.comparing(Event::getWhen, TimeRange.ORDER_BY_START));
+
+
+        TimeRange tempT = new TimeRange(0, 24 * 60);
+
+    /// if our duration is more than a day 
+        if(request.getDuration() > tempT.duration() ){
+            return availableTimes; // return empty array 
+        }
+        else if(request.getAttendees().size() == 0){
+                 availableTimes.add(tempT);
+                 return availableTimes;
+        }
+        
+        int tempConflict = 0;
+        int previous_endTime = 0; 
+        boolean conflict = false;
+  
+        ArrayList<String> currentAttendees = new ArrayList<String>();
+        ArrayList<TimeRange> availableGaps = new ArrayList<TimeRange>();
+        availableGaps = getGaps(eventsCopy);
+        
+
+
+
+
+        for(Event eV : eventsCopy){
+
+
+            for(String attendee: eV.getAttendees()){
```

Comment:
Instead of looping through all attendees, you just need to identify whether the event contains any of the attendees from the meeting request and thus might cause conflicts.

You can do that with the following:
```java
for (Event eV : eventsCopy) {
  If (!Collections.disjoint(eV.getAttendees(), request.getAttendees()) {
    ...
  }
}
```